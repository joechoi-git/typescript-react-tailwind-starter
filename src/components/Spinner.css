.spinner {
    height: 10vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .spinner {
        animation: spinner-keyframes infinite 20s linear;
    }
}

@keyframes spinner-keyframes {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
