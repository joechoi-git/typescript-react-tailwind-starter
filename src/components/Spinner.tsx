import React from "react";
import logo from "../logo.svg";
import "./Spinner.css";

export default function Spinner(): React.JSX.Element {
    return <img src={logo} className="spinner" alt="spinner" data-testid="spinner" />;
}
