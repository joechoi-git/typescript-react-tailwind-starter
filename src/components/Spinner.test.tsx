import React from "react";
import { render } from "@testing-library/react";
import Spinner from "./Spinner";
import logo from "../logo.svg";

describe("Spinner Component", () => {
    test("renders spinner image", () => {
        const { getByAltText } = render(<Spinner />);
        const spinnerImage = getByAltText("spinner");
        expect(spinnerImage).toBeInTheDocument();
        expect(spinnerImage).toHaveAttribute("src", logo);
        expect(spinnerImage).toHaveClass("spinner");
    });
});
