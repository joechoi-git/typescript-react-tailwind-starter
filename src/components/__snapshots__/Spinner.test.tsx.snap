// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="spinner"
    class="spinner"
    data-testid="spinner"
    src="test-file-stub"
  />
</DocumentFragment>
`;
